pkg/apis/work/v1alpha2/binding_types.go

```diff
@@ -362,6 +366,41 @@ type AggregatedStatusItem struct {
 	Health ResourceHealth `json:"health,omitempty"`
 }
 
+// FailoverHistoryItem represents either a failover event in the history.
+type FailoverHistoryItem struct {
+	// Reason denotes the type of failover.
+	// +required
+	Reason FailoverReason `json:"reason"`
+
+	// StartTime is the timestamp of when the failover occurred.
+	// +required
+	StartTime metav1.Time `json:"failoverTime"`
+
+	// ClustersBeforeFailover records the cluster where the application was running prior to failover.
+	// +required
+	ClusterBeforeFailover string `json:"originalCluster"`
+
+	// ClustersAfterFailover records the cluster where the application is running after failover.
+	// +optional
+	ClusterAfterFailover string `json:"targetCluster,omitempty"`
```

Comment:
I know in your case no matter before or after the failover, there is only one cluster. But I think we can do it commonly, that is record the clusters in a slice.  What do you think?

Reply:
I had noticed that the original definition was an array - does that mean we want to support failover for applications that are spread across multiple clusters? Or is the array simply going to contain a running history of the clusters on which is was scheduled before?

For example, let's say application X fails over from Cluster B to Cluster C -> the ClustersBeforeFailover will be: ["B"]. 

If application X later fails over from Cluster C to Cluster A -> the ClustersBeforeFailover will be ["B", "A"]?

Reply:
No. 
I just hope that the history information faithfully records the status, it doesn't care about how many clusters was scheduled before or after the failover. 

For your example, application X fails over from Cluster B to Cluster C, and then to Cluster A, then there will be two items on the history:
```yaml
- startTime: "Monday"
  clusterBeforeFailover: [B]
  clusterAfterFailover: [C]
- startTime: "Tuesday"
  clusterBeforeFailover: [C]
  clusterAfterFailover: [A]
```

Reply:
Ah understood,  thanks for clarification!